sacc2/data_types.py

```diff
@@ -0,0 +1,251 @@
+    "ggl_E": ["ell"],
+    "ggl_B": ["ell"],
+    
+
+}
+
+# This makes a namespace object, so you can do:
+# known_types.ggl_E == "ggl_E"
+# also, for convenience, you can do known_types.index('ggl_E') 
+# and 'ggl_E' in known_types
+
+known_types = Namespace(*required_tags.keys())
+
+
+class DataPoint:
+    """A class for a single data point (one scalar value).
+
+    Data points have a type, zero or more tracers, a value,
+    and any arbitrary tags that are stored in a dictionary,
+    and can be used to describe angular scales, window functions,
+    or any arbitrary information to be attached to the data.
+
+    Data points can be automatically created and added to a
+    Sacc object, so you don't normally nee to manually create them.
+
+    Attributes
+    -----------
+
+    data_type: str
+        A string, indicating the type of data point
+    tracers: tuple
+        Tuple of strings with the names of tracers to use
+    value: float
+        Mean value of this statistics
+    tags: dict
+        Dictionary of further data point metadata, such as binning info, angles, etc.
+
+    """
+    def __init__(self, data_type, tracers, value, ignore_missing_tags=False, **tags):
+        """Create a new data point.
+
+        Data points can be automatically created and added to a
+        Sacc object, so you don't normally nee to manually create them.
+        
+        Parameters
+        ----------
+
+        data_type: str
+            A string, indicating the type of data point
+        tracers: tuple
+            Tuple of strings with the names of tracers to use
+        value: float
+            Mean value of this statistics
+        ignore_missing_tags: bool
+            Optional, default=False.  If True, do not complain if a tracer usually
+            needed for this data type is not present.
+        **tags: dict[str:any]
+            Dictionary of further data point metadata, such as binning info, angles, etc.
+        """
+        self.data_type = data_type
+        self.tracers = tracers
+        self.value = value
+        self.tags = tags
+        # Data types can have required tags which must be present.
+        # Check for those here
+        if (data_type in required_tags) and (not ignore_missing_tags):
+            for tag in required_tags[data_type]:
+                if tag not in tags:
+                    raise ValueError(f"Tag {tag} required for data type {data_type} (ignore_missing_tags=False)")
+
+
+        # We encourage people to use existing type names, and issue a warning if they do
+        # not to prod them in the right direction.
+        # We are removing this warning until we converge on what the data types should be
+        # if data_type not in known_types:
+        #     warnings.warn(f"Unknown data_type value {data_type}. If possible use a pre-defined type, or add to the list.")
+
+    def __repr__(self):
+        return f"<Data {self.data_type} {self.tracers} {self.value} {self.tags}>s"
```

Comment:
Nice - done.  It won't work 100% (because it would be impractical to make window function objects eval in the same way, since they can be complicated) but it should mostly work.